### PR TITLE
Add support for socket

### PR DESCRIPTION
**Pull request by yabirgb**
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

### CODE REVIEW

File: rotkehlchen/chain/evm/decoding/socket_bridge/constants.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from typing import Final
+
+from rotkehlchen.chain.evm.types import string_to_evm_address
+
+CPT_SOCKET: Final = 'socket'
+GATEWAY_ADDRESS: Final = string_to_evm_address('0x3a23F943181408EAC424116Af7b7790c94Cb97a5')
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/socket_bridge/constants.py:6]: Is it the same in every EVM chain? That's weird.

File: docs/changelog.rst [hunk 2]
```diff
@@ -10,6 +10,7 @@ Changelog
 * :feature:`7146` The exported CSV for PnL Report now contains an Asset column with symbols.
 * :feature:`6254` Users can now stop the execution of long-running queries.
 * :feature:`7092` Users of metamask swaps will now see them properly decoded in the history view and have them taken into account during accounting.
+* :feature:`-`  Add support for bridge events initialized using Socket.
```
Comment (LefterisJP) [inline docs/changelog.rst:13]:
```suggestion
* :feature:`-`  Bridging transactions across all EVM chains done via socket will now be decoded properly.
```

File: rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        sender = hex_or_bytes_to_address(context.tx_log.data[128:160])
+        receiver = hex_or_bytes_to_address(context.tx_log.data[160:192])
+        to_chain_id_raw = hex_or_bytes_to_int(context.tx_log.data[64:96])
+
+        try:
+            to_chain_id = ChainID.deserialize_from_db(to_chain_id_raw)
+            _, target_chain = to_chain_id.name_and_label()
+        except DeserializationError:
+            target_chain = str(to_chain_id_raw)
+            log.error(f'Unknown chain in socket bridge {to_chain_id_raw}')
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py:78]:
```suggestion
            log.error(f'Unknown to_chain in socket bridge: {to_chain_id_raw}')
```

File: rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+                event.event_type == HistoryEventType.SPEND
+            ):
+                if self.base.is_tracked(receiver):  # if receiver is not tracked we are spending it  # noqa: E501
+                    event.event_type = HistoryEventType.WITHDRAWAL
+                    event.event_subtype = HistoryEventSubType.BRIDGE
+
+                event.counterparty = CPT_SOCKET
+                _, from_chain = self.evm_inquirer.chain_id.name_and_label()
+                event.notes = (
+                    f'Bridge {amount} {bridged_asset.symbol} from {from_chain} '
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py:94]: If this is always from current chain to another, by saying to you are covered. You see the current chain already in the transaction and the event itself. No need to repeat here.
```suggestion
                    f'Bridge {amount} {bridged_asset.symbol} '
```

File: rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                    f'to {receiver} at {target_chain} using Socket'
+                )
+
+        return DEFAULT_DECODING_OUTPUT
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {
+            GATEWAY_ADDRESS: (self._decode_bridged_asset,),
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py:104]: This seems to only handle bridging to another chain. Can you also handle the bridge receiving (withdrawal) case?

File: rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+        for event in context.decoded_events:
+            if (
+                event.location_label == sender and
+                event.asset == bridged_asset and
+                event.balance.amount == amount and
+                event.event_type == HistoryEventType.SPEND
+            ):
+                if self.base.is_tracked(receiver):  # if receiver is not tracked we are spending it  # noqa: E501
+                    event.event_type = HistoryEventType.WITHDRAWAL
```
Comment (LefterisJP) [inline rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py:88]: Shouldn't it be a DEPOSIT? This is what we use for sending/depositing to a bridge. And from what I see, this is for sending to a bridge no?

```suggestion
                    event.event_type = HistoryEventType.DEPOSIT
```

File: rotkehlchen/chain/evm/decoding/socket_bridge/constants.py (hunk 1 above)
Comment (yabirgb) [reply rotkehlchen/chain/evm/decoding/socket_bridge/constants.py:6]: Yes, it is stated here https://docs.socket.tech/socket-api/contract-addresses

File: rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py (hunk 5 above)
Comment (yabirgb) [reply rotkehlchen/chain/evm/decoding/socket_bridge/decoder.py:102]: As discussed in the standup it delegates the bridging to other sources and those handle the withdrawal

**Review by LefterisJP — APPROVED** [review]
LGTM

**Pull request merged**